File: 102. Binary Tree Level Order Traversal/binaryTreeLevelOrderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

#include <queue>
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == NULL) return vector<vector<int> >();
        vector<vector<int> > result;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            vector<int> level;
            queue<TreeNode*> temp(q);
            while(!temp.empty()){
                TreeNode* node = temp.front();
                temp.pop();
                q.pop();
                level.push_back(node->val);
                if(node->left != NULL)
                    q.push(node->left);
                if(node->right != NULL)
                    q.push(node->right);
            }
            result.push_back(level);
        }
        return result;
    }
};

File: 102. Binary Tree Level Order Traversal/binaryTreeLevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

from collections import deque

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root == None:
            return []
        q = deque()
        q.append(root)
        result = []
        while q:
            level = []
            temp = q.copy()
            while temp:
                e = temp.pop()
                q.pop()
                level.append(e.val)
                if e.left != None:
                    q.appendleft(e.left)
                if e.right != None:
                    q.appendleft(e.right)
            result.append(level)
        return result
                    
